pleque/core/equilibrium.py

```diff
@@ -1514,6 +1514,130 @@ def connection_length(self, *coordinates, R: np.array = None, Z: np.array = None
 
         return dists, lines
 
+    def trace_field_line_toroidal(self, *coordinates, R: np.array = None, Z: np.array = None,
+                         coord_type=None, in_first_wall=True, phi_end = None, phi_eval = None, **coords):
+        """
+        Return traced field lines starting from the given set of at least 2d coordinates.
+        No stopper method are used here, the toroidal angle interval is specified by the user.
+        :param coordinates: coordinates at the start of the field lines. If 2D dimensionned, start at phi = 0
+        :param R:
+        :param Z:
+        :param coord_type: 
+        :param direction: if positive trace field line in/cons the direction of magnetic field.
+        :param in_first_wall: if True the only inner part of field line is returned.
+        :param phi_end : toroidal angle value at wich to stop the integration. 
+        :param phi_eval : optional, allow to choose the values of the toroidal angle returned 
+            along the integration of the magnetic field line. 
+            Must be contained in the range [phi_start, phi_end], the minimum and maximum value of the toroidal angle.
+            If none specified, allow the solver to choose these values. 
+        :param coords:
+        :return: list of coordinates 
+        """
+
+
+        coords = self.coordinates(*coordinates, R=R, Z=Z, coord_type=coord_type, **coords)
+
+        res = []
+
+        # XXXNOW
+        coords_rz = coords.as_array(dim=2)
+
+
+
+        dphifunc = flt.dphi_tracer_factory(self.B_R, self.B_Z, self.B_tor)
+
+        r_lims = [np.min(self.first_wall.R), np.max(self.first_wall.R)]
+        z_lims = [np.min(self.first_wall.Z), np.max(self.first_wall.Z)]
+
+        for i in np.arange(len(coords)):
+
+            if 1: #i==0 or i==len(coords)-1 or i == np.int(len(coords)/2):
+                y0 = coords_rz[i]
+                if coords.dim == 2:
+                    phi_start = 0
+                else:
+                    phi_start = coords.phi[i]
+
+                if phi_end is None:
+                    phi_end = phi_start 
```

Comment:
If you set this, it will do nothing. You may require at least one toroidal orbit. See original function. There is a formula to resolve the direction.